modules/apps/transfer/keeper/msg_server.go

```diff
@@ -36,6 +36,11 @@ func (k Keeper) Transfer(goCtx context.Context, msg *types.MsgTransfer) (*types.
 		return nil, errorsmod.Wrapf(ibcerrors.ErrUnauthorized, "%s is not allowed to send funds", sender)
 	}
 
+	// Initialize empty pointer with empty forwarding information
+	if msg.Forwarding == nil {
+		msg.Forwarding = types.NewForwarding(false)
+	}
```

Comment:
Why is this needed? 

If it's only for the check below (and for the call to `msg.Forwarding.Hops`, maybe it can be *slightly* cleaner like this?

```go
var hops []types.Hops
if msg.Forwarding  != nil {
      hops = msg.Forwarding.Hops 
      if msg.Forwarding.Unwind {
         ....
       } 
}

	sequence, err := k.sendTransfer(
		ctx, msg.SourcePort, msg.SourceChannel, coins, sender, msg.Receiver, msg.TimeoutHeight, msg.TimeoutTimestamp,
		msg.Memo, hops)
```

Still not super clean, but maybe marginally better?

Reply:
maybe also pretty dirty, but could do something like
```golang
func (f *Forwarding) Unwind() {
  if f == nil {
    return false
  }
  return f.Unwind 
}
```

Perfectly happy with the inline check colin suggested though :+1: